okhttp-ws-tests/src/test/java/com/squareup/okhttp/internal/ws/RealWebSocketTest.java

```diff
@@ -207,7 +219,92 @@
     }
   }
 
-  @Test public void serverCloseThenWritingThrows() throws IOException {
+  @Test public void clientWritingThrowsSendsClientClose() throws IOException {
+    final IOException brokenException = new IOException("Broken!");
+    RequestBody brokenBody = new RequestBody() {
+      @Override public MediaType contentType() {
+        return TEXT;
+      }
+
+      @Override public void writeTo(BufferedSink sink) throws IOException {
+        throw brokenException;
+      }
+    };
+
+    try {
+      client.sendMessage(brokenBody);
+      fail();
+    } catch (IOException e) {
+      assertSame(brokenException, e);
+    }
+
+    // A failed write prevents further use of the WebSocket instance.
+    try {
+      client.sendPing(new Buffer().writeUtf8("Ping!"));
+      fail();
+    } catch (IllegalStateException e) {
```

Comment:
There is only one client. If it gets IO then subsequent interactions are a
programming bug.

On Sat, Nov 14, 2015, 9:34 AM Jesse Wilson <notifications@github.com> wrote:

> In
> okhttp-ws-tests/src/test/java/com/squareup/okhttp/internal/ws/RealWebSocketTest.java
> <https://github.com/square/okhttp/pull/1996#discussion_r44859075>:
>
> > +        throw brokenException;
> > +      }
> > +    };
> > +
> > +    try {
> > +      client.sendMessage(brokenBody);
> > +      fail();
> > +    } catch (IOException e) {
> > +      assertSame(brokenException, e);
> > +    }
> > +
> > +    // A failed write prevents further use of the WebSocket instance.
> > +    try {
> > +      client.sendPing(new Buffer().writeUtf8("Ping!"));
> > +      fail();
> > +    } catch (IllegalStateException e) {
>
> not sure I like the consequences here. One client throwing an IOException
> can cause another client to get a runtime exception?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/1996/files#r44859075>.
>